astropy/config/paths.py

```diff
@@ -7,11 +7,12 @@
 import shutil
 import sys
 from functools import wraps
+from pathlib import Path
 
 __all__ = ["get_config_dir", "get_cache_dir", "set_temp_config", "set_temp_cache"]
 
 
-def _find_home():
+def _find_home() -> Path:
```

Comment:
It's not at all obvious to me why we need this function at all. Why couldn't we just use [`Path.home()`](https://docs.python.org/3/library/pathlib.html#pathlib.Path.home)? The vast majority of this function is for Windows, so @pllim needs to have a look at this.

Reply:
yeah, codecov seems useful for once and shows that none of the `except` block is used in any test.

Reply:
Blame says this code was added at least 13 years ago (https://github.com/astropy/astropy/pull/73), so probably predated a lot of Path stuff and maybe even `os.path` was incomplete back then.

Reply:
I am okay with removing it but should probably be a separate PR. Unless you want to do that first and then come back to this PR? 🤷‍♀️

Reply:
> Unless you want to do that first and then come back to this PR? 🤷‍♀️

It makes more sense to me. Let me open a PR now.

Reply:
done

Reply:
Removal of legacy things like this tend to cause havoc downstream, so it would be superb to do these removals more carefully, and at least mention them in the changelog as it would save time for those who try to investigate why packages became incompatible with the latest release (specifics are not really important, but this time you broke halotools).

Reply:
There's no good reason why changes in the private API should be mentioned in the change log at all.

Reply:
xref for future case studies: https://github.com/astropy/halotools/pull/1107